clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10263,8 +10263,18 @@ void OffloadWrapper::ConstructJob(Compilation &C, const JobAction &JA,
     const InputInfo &I = Inputs[0];
     assert(I.isFilename() && "Invalid input.");
 
-    if (I.getType() == types::TY_Tempfiletable ||
-        I.getType() == types::TY_Tempfilelist || IsEmbeddedIR)
+    // TODO: The embedded compilation step after the wrapping step restricts
```

Comment:
Why is this a TODO? Is this not more of a restriction?

Reply:
It is a restriction - but in my opinion it would be best to pull the compile step out of the wrapping step which provides a cleaner picture of the toolchain.  We shouldn't expect the wrapping step to do the compilation too (it isn't embedded for OpenMP)